cloud/linode/cilium_loadbalancers.go

```diff
@@ -0,0 +1,445 @@
+package linode
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2alpha1"
+	ciliumclient "github.com/cilium/cilium/pkg/k8s/client/clientset/versioned/typed/cilium.io/v2alpha1"
+	slimv1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/apis/meta/v1"
+	"github.com/google/uuid"
+	"github.com/linode/linode-cloud-controller-manager/cloud/annotations"
+	"github.com/linode/linodego"
+	v1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/retry"
+	"k8s.io/klog/v2"
+	"k8s.io/utils/ptr"
+)
+
+const (
+	ciliumLBClass              = "io.cilium/bgp-control-plane"
+	ipHolderLabelPrefix        = "linode-ccm-ip-holder"
+	ciliumBGPPeeringPolicyName = "linode-ccm-bgp-peering"
+)
+
+// This mapping is unfortunately necessary since there is no way to get the
+// numeric ID for a data center from the API.
+// These values come from https://www.linode.com/docs/products/compute/compute-instances/guides/failover/#ip-sharing-availability
+var (
+	regionIDMap = map[string]int{
+		"us-southeast": 4,  // Atlanta, GA (USA)
```

Comment:
can we move this to a configmap as a followup?

Reply:
How would a configmap help? I think we'd still need to cut a new release if we did add/remove a DC, it would just be a difference of if the image version changes vs the k8s resources

Reply:
I guess i was thinking more from a internal customer pov - modifying a configmap in place vs picking up a new version - but you're right, doesn't give us much